Add QGIS first tutorial in Ecology topic (#4381)
galaxyproject/training-material: topics/ecology/tutorials/QGIS_Web_Feature_Services/tutorial.md

---
layout: tutorial_hands_on
title: QGIS Web Feature Services
questions:
- How to use WFS webservice to access and import GIS data through Galaxy
objectives:
- Deal with QGIS execution management
- Access, filter, and import GIS data using WFS webservice
time_estimation: 0H30M
key_points:
- From WFS webservice, you can now access GIS remote data, build a query to select a subpart of the data, and import it to Galaxy through QGIS Galaxy interactive tool.
tags:
  - GIS
  - Geographical Information System
  - WFS
  - Spatial data
  - QGIS
  - Maps
  - OGC
contributors:
- colineroyaux
- Marie59
- yvanlebras

---

# Introduction

Based on this [QGIS official tutorial](https://docs.qgis.org/2.18/en/docs/training_manual/online_resources/wfs.html), you will learn here how to access, filter and import GIS data through WFS web service using QGIS Galaxy interactive tool:

> In the Geographical Information System landscape, there is existing standards to help users deal with remote data. The most common web services are Web Map Services (WMS) and Web Feature Services (WFS). If WMS allows users only to access and display maps stored remotely, WFS is giving access to the features of data so you can modify it and create your own data and maps.

Reviewer: https://training.galaxyproject.org/training-material/news/2021/05/25/abbreviations-tag.html might be useful here

---
layout: tutorial_hands_on

> 

Reviewer 2: ```suggestion
```

title: QGIS Web Feature Services
questions:

> - how to use WFS webservice to access and import GIS data through Galaxy

Reviewer 2: ```suggestion
- How to use WFS webservice to access and import GIS data through Galaxy
```

objectives:
- Deal with QGIS execution management
- Access, filter, and import GIS data using WFS webservice
time_estimation: 0H30M
key_points:
- From WFS webservice, you can now access GIS remote data, build a query to select a subpart of the data, and import it to Galaxy through QGIS Galaxy interactive tool.
tags:
  - GIS
  - Geographical Information System
  - WFS
  - Spatial data
  - QGIS
  - Maps
  - OGC
contributors:
- colineroyaux
- Marie59
- yvanlebras

---

# Introduction

> 

Reviewer 2: ```suggestion
```

> <hands-on-title>Deploy your own QGIS instance</hands-on-title>
>
> 1. Create a new history for this tutorial
> 2. Deploy a {% tool [QGIS instance](https://ecology.usegalaxy.eu/root?tool_id=interactive_tool_qgis) %}
> 3. Access QGIS
> 4. Save your project in /working/output folder
>
{: .hands_on}

# Web Feature Services
## Loading WFS layer

> <hands-on-title>Loading WFS layer</hands-on-title>
>
> 1. Go to the top left of QGIS and click on "Layer"
> 2. Click on the 1st option "Data Source Manager" 
> 3. Select "WFS / OGC API - Features" on left part of the pop-up panel
> 4. Click the "New" button
> 5. In the dialog that appears, enter the Name as `nsidc` and the URL as `http://nsidc.org/cgi-bin/atlas_south?version=1.1.0`.
>
>    > <tip-title>copy pasting from computer to QGIS</tip-title>
>    >
>    > * You can expand the QGIS left panel (where there is 3 dots, verticaly) to access the "clipboard" menu, and paste the content you want to paste on a QGIS form. Then, click outside of this panel to collapse it, and you can click for example on the `url` field to paste the url from your clipbaord
>    >

> >    > ![QGIS clipboard](../../images/QGIS/qgistuto1.PNG)

Reviewer: It looks like this alt-text doesn't describe the image very well for users who are relying on a screen reader to access the content. We have a short guide on [what's appropriate for alt-text vs captions in the GTN Contributing Guide](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#guidelines-on-alt-vs-figcaption-text) which has more details. *This comment may apply to more images, please consider checking all of them.*

> 2. Go to the top left of QGIS and click on "Layer"
> 3. Click on the 1st option "Data Source Manager" 
> 4. Select "WFS / OGC API - Features" then the already connected `nsidc` server
> 5. Select the `countries (excluding Antarctica)` layer and click "Build query" button
>    > ![Image of QGIS interface to help you see which layers to select in the WFS / OGC API features](../../images/QGIS/qgistuto8.PNG)
> 6. On the new dialog box, you can copy/paste this query: `SELECT * FROM country_borders_excluding_antarctica WHERE "Countryeng" = 'South Africa'` using the clipboard functionnality from the left QGIS panel to make a "bridge" between your local clipboard and the virtualized QGIS one.
>    > ![Image of QGIS interface to help you see what to write in the "SQL query composer"](../../images/QGIS/qgistuto9.PNG)
> 7. Clicking ok, you can now see the SQL query on a dedicated column of the layers table.
>    > ![Image of QGIS interface showing you the results of the SQL command on the selection of layers in the WFS / OGC API features](../../images/QGIS/qgistuto10.PNG)
> 9. Clicking the "Add" button, you now have the South Africa layer added and diplayed on QGIS.

> >    > ![Image of QGIS interface showing the resulting map of all your changes since the beginning of the tutorial](../../images/QGIS/qgistuto11.PNG)

Reviewer: ```suggestion
>    ![Image of QGIS interface showing the resulting map of all your changes since the beginning of the tutorial](../../images/QGIS/qgistuto11.PNG)
```

please remove the `>` that are in front of the images, they mark the beginning of a block quote and are unnecessary.

Reviewer: (This applies to all of the imgaes it seems, if you're in the "Files" https://github.com/galaxyproject/training-material/pull/4381/files tab you'll see the linting error messages about a potentially broken box.

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Managing QGIS Galaxy interactive tool

QGIS is now integrated in Galaxy as an interactive tool. This kind of tool works differently than classical tools as it allows the user to interact with a dedicated graphical interface.

To use QGIS, you need to use the {% tool [dedicated form](interactive_tool_qgis) %}, you can specify input datasets from your history you want to use in QGIS, or not ;), then press the execute button to launch a QGIS instance. When the graphical user interface of QGIS is ready to be used, a URL will be displayed at the top of the Galaxy center panel. If you don't see it, you can see and access it through the "Active InteractiveTools" space of the "User" menu or you can click on {% icon galaxy-eye %} on the tool in the history.

> Once you finish your work on QGIS, if you want to retrieve data and/or the entire project, you need to save files in /output, then quit QGIS properly through the "Project" Menu tab.

Reviewer 2: ```suggestion
Once you finish your work on QGIS, if you want to reuse data and/or the entire project, you need to save files in the "output" folder, then quit QGIS properly through the "Project" Menu tab.
```

Reviewer 2: Maybe we need to describe where the output folder can be found.

Reviewer 2: Below we say /working/output for example.

>    ![Image of QGIS interface to help you see what to write in the "SQL query composer"](../../images/QGIS/qgistuto9.PNG)
> 7. Clicking OK, you can now see the SQL query on a dedicated column of the layers table.
>    ![Image of QGIS interface showing you the results of the SQL command on the selection of layers in the WFS / OGC API features](../../images/QGIS/qgistuto10.PNG)
> 9. Clicking the "Add" button, you now have the South Africa layer added and displayed on QGIS.
>    ![Image of QGIS interface showing the resulting map of all your changes since the beginning of the tutorial](../../images/QGIS/qgistuto11.PNG)
{: .hands_on}

> <question-title></question-title>
>
> 1. Are you seeing how QGIS shows the fact that a layer is resulting from a query made on a larger layer?
>
> > <solution-title></solution-title>
> >
> > 1. There is a dedicated "filter" icon next to the name of the layer.
> >
> {: .solution}
>
{: .question}

> 
> 
> 
> 
> 
> 

Reviewer 2: ```suggestion
```

# Conclusion

> 

Reviewer 2: ```suggestion
```
